version: "3.9"

services:
  db:
    image: postgres:12-bullseye
    container_name: postgres_db
    environment:
      POSTGRES_USER: storage_admin
      POSTGRES_PASSWORD: THw7l0bxvPPkWUhP
      POSTGRES_DB: strg_users_db
    ports:
      - "5435:5432" #
    volumes:
      - ./docker-data/postgres:/var/lib/postgresql/data
      - ./db_init/init.sql:/docker-entrypoint-initdb.d/init.sql
    networks:
      - app_network
    deploy:
      resources:
        limits:
          cpus: '0.3'
          memory: 512MB
      restart_policy:
        condition: on-failure
        max_attempts: 3
        delay: 5s
        window: 20s

  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: python_app
    volumes:
      - ./logs:/app/logs
    depends_on:
      - db
    environment:
      DATABASE_URL: postgresql://storage_admin:THw7l0bxvPPkWUhP@db:5432/strg_users_db
    ports:
      - "8000:8000"
    networks:
      - app_network
    deploy:
      resources:
        limits:
          cpus: '0.3'
          memory: 512MB
      restart_policy:
        condition: on-failure
        max_attempts: 3
        delay: 5s
        window: 20s

volumes:
  postgres_data:


networks:
  app_network:
    driver: bridge
